mm2src/coins/eth/fee_estimation/eth_fee_events.rs

```diff
@@ -0,0 +1,93 @@
+use super::ser::FeePerGasEstimated;
+use crate::eth::EthCoin;
+use common::executor::Timer;
+use mm2_event_stream::{Broadcaster, Event, EventStreamer, NoDataIn, StreamHandlerInput};
+
+use async_trait::async_trait;
+use futures::channel::oneshot;
+use instant::Instant;
+use serde::Deserialize;
+use std::convert::TryFrom;
+
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields)]
+/// Types of estimators available.
+/// Simple - simple internal gas price estimator based on historical data.
+/// Provider - gas price estimator using external provider (using gas api).
+pub enum EstimatorType {
+    Simple,
+    Provider,
+}
+
+#[derive(Deserialize)]
+#[serde(deny_unknown_fields, default)]
+pub struct EthFeeStreamingConfig {
+    /// The time in seconds to wait before re-estimating the gas fees.
+    pub estimate_every: f64,
+    /// The type of the estimator to use.
+    pub estimator_type: EstimatorType,
+}
+
+impl Default for EthFeeStreamingConfig {
+    fn default() -> Self {
+        Self {
+            estimate_every: 15.0,
```

Comment:
I think this param should not be user-defined. For the simple provider it should be run not more than once in blocktime.
Other providers allow to get more dynamic gas estimations (based on txns in mempool), so could be call with less interval, but I think this is up to the team to define this interval. I think it should be taken from KDF config file.